ui/learn/src/map/mapSide.ts

```diff
@@ -89,19 +90,76 @@ function renderHome(ctrl: SideCtrl) {
         },
       }),
     ]),
-    m('div.actions', [
+    m(
+      'div.actions',
       progress > 0
-        ? m(
-            'a.confirm',
-            {
-              onclick: function () {
-                if (confirm(ctrl.trans.noarg('youWillLoseAllYourProgress'))) ctrl.reset();
+        ? [
+            m(
+              'label',
+              {
+                for: 'reset-categ-dropdown',
               },
-            },
-            ctrl.trans.noarg('resetMyProgress')
-          )
-        : null,
-    ]),
+              ctrl.trans.noarg('resetMyProgress')
+            ),
+            m(
+              'select',
+              {
+                name: 'reset-categ-dropdown',
+                id: 'reset-categ-dropdown',
+                oninput: (value: Event | null) => {
+                  const target = value?.target as HTMLSelectElement;
+                  const question = target.selectedOptions[0].getAttribute('question');
+                  if (question) {
+                    if (target.value !== '' && confirm(question)) {
+                      if (target.value === 'ALL') {
+                        ctrl.reset();
+                      } else {
+                        ctrl.partial_reset(target.value);
+                      }
+                    }
+                    target.value = '';
+                  }
+                },
+              },
+              ...[
+                ...[
+                  m(
+                    'option',
+                    {
+                      value: '',
+                    },
+                    ctrl.trans.noarg('choose')
+                  ),
+                  m(
+                    'option',
+                    {
+                      value: 'ALL',
+                      question: ctrl.trans.noarg('youWillLoseAllYourProgress'),
+                    },
+                    ctrl.trans.noarg('everything')
+                  ),
+                ],
+                ...[
+                  ...stages.categs.map(function (categ) {
+                    return categ.stages.map(function (s) {
```

Comment:
Does this actually work properly? Wouldn't you get nested lists? Or does mithril unwrap them automatically or something? Also would use arrow functions here.

Reply:
Yes this worked, if not I wouldn't have commited this. I needed the 4x "..." weirdly. I tried for a long time to remove them, to no avail. I would consider arrow functions, but then isn't there a reason it is made in a longer form above ? Shouldn't I keep the same way it is made ?